packages/react-composites/tests/browser/callwithchat/CallWithChatComposite.test.ts

```diff
@@ -70,6 +70,48 @@ test.describe('CallWithChat Composite CallWithChat Page Tests', () => {
     expect(await pages[0].screenshot()).toMatchSnapshot(`call-with-chat-gallery-screen-with-chat-pane.png`);
   });
 
+  test('Unread chat message button badge are displayed correctly for <9 messages', async ({ pages }) => {
+    // Open chat pane on page 0 and send a message
+    await pageClick(pages[0], dataUiId('call-with-chat-composite-chat-button'));
+    await waitForSelector(pages[0], dataUiId('call-with-chat-composite-chat-pane'));
+    await sendMessage(pages[0], 'Call with Chat composite is awesome!');
+    await waitForMessageDelivered(pages[0]);
+
+    // Ensure typing indicator has disappeared to prevent flakey test
+    await pages[1].bringToFront();
+    const typingIndicator = await pages[1].$(dataUiId(IDS.typingIndicator));
+    typingIndicator && (await typingIndicator.waitForElementState('hidden'));
+    await waitForSelector(pages[1], dataUiId('call-with-chat-composite-chat-button-unread-icon'));
+    expect(await pages[1].screenshot()).toMatchSnapshot(`call-with-chat-gallery-screen-with-one-unread-messages.png`);
+  });
+
+  test('Unread chat message button badge are displayed correctly for >9 messages', async ({ pages }) => {
+    // Open chat pane on page 0 and send 10 messages
+    await pageClick(pages[0], dataUiId('call-with-chat-composite-chat-button'));
+    await waitForSelector(pages[0], dataUiId('call-with-chat-composite-chat-pane'));
+
+    for (let i = 0; i < 10; i++) {
+      await sendMessage(pages[0], 'Call with Chat composite is awesome!');
+      // timeout between each messages to prevent chat throttling
+      await pages[0].waitForTimeout(100);
+    }
+
+    // Ensure typing indicator has disappeared to prevent flakey test
+    await pages[1].bringToFront();
+    const typingIndicator = await pages[1].$(dataUiId(IDS.typingIndicator));
+    typingIndicator && (await typingIndicator.waitForElementState('hidden'));
+
+    await waitForMessageSeen(pages[0]);
```

Comment:
Is this `waitForMessagesSeen` function supposed to be about the messages being seen by the recipient of the messages? or is this one about the message sent indicator?

Reply:
In this test even though page[1] didn't see the message, the messages sent by pages[0] arrives at the seen state. This waitForMessagesSeen is just another layer of prevention against flaky tests to make sure there are seen icons in page[0]